lib/Raven/Client.php

```diff
@@ -423,8 +436,8 @@ public function capture(array $payload)
 
         if (isset($payload['culprit'])) {
             $event = $event->withCulprit($payload['culprit']);
-        } else {
-            $event = $event->withCulprit($this->transaction->peek());
+        } elseif (!$this->transactionStack->isEmpty()) {
```

Comment:
This is one of the reasons for which I don't agree with throwing an exception with an empty stack: you are forced to add additional logic even inside the client.
An empty stack is not an "error", so IMHO an exception is not warranted.

Reply:
Java throws an [exception](https://docs.oracle.com/javase/10/docs/api/java/util/EmptyStackException.html) too if trying to access a stack that is empty. For sure it's more tedious to have to wrap everything in a `if` condition, but I think that the intent is more clear this way. As an example, what if the stack could contain any value `null` included? Throwing an exception if the stack is empty makes clear that there is something wrong while returning `null` could be misleading. I still think that even if it's not possible here due to the class only accepting strings we should do it this way

Reply:
I'll give you one solid reason: https://github.com/getsentry/sentry-php/pull/574/files#diff-6d6ad8fef701ebc6ede525e4b6c8f8d0R153

I just merged a PR on master to handle exceptions without stack that we provoke ourselves.

Reply:
>I just merged a PR on master to handle exceptions without stack that we provoke ourselves.

How could the exception I throw be without stack?

>returning null could be misleading

Although this cannot happen here due to the code that prevents non-string types to be pushed to the stack, a good code would not return null as it could be a value conained in the stack and an end-user would no longer be able to distinguish if the stack is empty or if the returned value has been really peeked/popped from it